tests/components/mikrotik/test_hub.py

```diff
@@ -77,22 +76,28 @@ def mock_command(self, cmd, params=None):
 async def test_hub_setup_failed(hass):
     """Failed setup of Mikrotik hub."""
 
+    config_entry = MockConfigEntry(domain=mikrotik.DOMAIN, data=MOCK_DATA)
+    config_entry.add_to_hass(hass)
     # error when connection fails
     with patch(
         "librouteros.connect", side_effect=librouteros.exceptions.ConnectionError
-    ), pytest.raises(ConfigEntryNotReady):
-        config_entry = MockConfigEntry(domain=mikrotik.DOMAIN, data=MOCK_DATA)
-        await mikrotik.async_setup_entry(hass, config_entry)
+    ):
+
+        await hass.config_entries.async_setup(config_entry.entry_id)
+
+        assert config_entry.state == config_entries.ENTRY_STATE_SETUP_RETRY
 
     # error when username or password is invalid
     with patch(
         "homeassistant.config_entries.ConfigEntries.async_forward_entry_setup"
     ) as forward_entry_setup, patch(
         "librouteros.connect",
         side_effect=librouteros.exceptions.TrapError("invalid user name or password"),
+    ), pytest.raises(
+        config_entries.OperationNotAllowed
```

Comment:
This shouldn't happen unless we have a bug.

Reply:
this is happening because if the username or password are wrong it will return false instead of ConfigEntryNotReady. As per your request using `await hass.config_entries.async_setup(config_entry.entry_id)` requires adding to hass, thus we get this error.

Reply:
No, this is happening cause we are trying to set up the same config entry twice without unloading it in between. That's not allowed. Create a new mock entry and test setting it up with the auth failure.

Reply:
Thanks for the help. Resolved